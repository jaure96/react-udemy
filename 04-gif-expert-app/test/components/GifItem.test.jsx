import { render, screen } from "@testing-library/react";
import { GifItem } from "../../src/components/GifItem";
describe("Pruebas en <GifItem />", () => {
  const title = "A simple title";
  const url = "http://one-punch.com/saitama.jpg";

  test("Had to match with the snapshot", () => {
    const { container } = render(<GifItem title={title} url={url} />);
    expect(container).toMatchSnapshot();
  });

  test("It should show the url and alt with the given image", () => {
    render(<GifItem title={title} url={url} />);
    const { src, alt } = screen.getByRole("img");
    expect(src).toBe(url);
    expect(alt).toBe(title);
  });

  test("It should show the title in the component", () => {
    render(<GifItem title={title} url={url} />);
    expect(screen.getByText(title)).toBeTruthy();
  });
});
