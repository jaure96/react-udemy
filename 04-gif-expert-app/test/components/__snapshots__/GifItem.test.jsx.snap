// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <GifItem /> Ha to match with the snapshot 1`] = `
<div>
  <div
    class="card"
  >
    <img
      alt="A simple title"
      src="http://one-punch.com/saitama.jpg"
    />
    <p>
      A simple title
    </p>
  </div>
</div>
`;

exports[`Pruebas en <GifItem /> Had to match with the snapshot 1`] = `
<div>
  <div
    class="card"
  >
    <img
      alt="A simple title"
      src="http://one-punch.com/saitama.jpg"
    />
    <p>
      A simple title
    </p>
  </div>
</div>
`;

exports[`Test in <GifItem /> Had to match with the snapshot 1`] = `
<div>
  <div
    class="card"
  >
    <img
      alt="A simple title"
      src="http://one-punch.com/saitama.jpg"
    />
    <p>
      A simple title
    </p>
  </div>
</div>
`;
